iree/compiler/Dialect/VM/Target/C/CModuleTarget.cpp

```diff
@@ -151,45 +201,129 @@ static LogicalResult translateFunctionToC(mlir::emitc::CppEmitter &emitter,
 }
 
 static LogicalResult buildModuleDescriptors(IREE::VM::ModuleOp &moduleOp,
-                                            llvm::raw_ostream &output) {
+                                            mlir::emitc::CppEmitter &emitter) {
+  SymbolTable symbolTable(moduleOp);
   std::string moduleName = moduleOp.getName().str();
+  llvm::raw_ostream &output = emitter.ostream();
+
+  if (failed(printStructDefinitions(moduleOp, output))) {
+    return failure();
+  }
+
+  // function wrapper
+  for (auto funcOp : moduleOp.getOps<IREE::VM::FuncOp>()) {
+    output << "static iree_status_t "
+           << buildFunctionName(moduleOp, funcOp,
+                                /*implSufffix=*/false)
+           << "("
+           << "iree_vm_stack_t* stack, " << moduleName << "_t* module, "
+           << moduleName << "_state_t* module_state";
+
+    if (funcOp.getNumArguments() > 0) {
+      output << ", ";
+    }
+
+    if (failed(printFuncOpArguments(funcOp, emitter))) {
+      return failure();
+    }
+
+    if (funcOp.getNumResults() > 0) {
+      output << ", ";
+    }
+
+    std::vector<std::string> resultNames;
```

Comment:
(whether useful or not) the LLVM style is to almost always use SmallVector instead of std::vector - not critical in our code but useful to keep in sync with their guidelines - so this would be `SmallVector<std::string, 4>` (or something - AFAICT it's a "guess the size" kind of thing)

Reply:
Sure, I just picked the first container that came to mind :laughing: